### DISCUSSION TITLE
How to zoom elements in sheet view?

### DISCUSSION THREAD

**Discussion by fatalstar** [iTwin/itwinjs-core#7439] [Q&A] [answered]
Hi, I opened a 2D sheet and found an element with ID 0x2000000130c, but I cannot zoom it or changeColor in this SheetView. 
![2024-12-02T10_03_10](https://github.com/user-attachments/assets/4261dc7e-0f28-4231-a362-6ecc13c6fc29)

I found the element is in the drawing model, and if I load the drawing, the zoom or changeColor in DrawingView will succeed.

**Comment by pmconne** [comment, marked as answer]
The drawing and sheet are two different models with two different coordinate systems. The appearance of the geometry in the drawing when viewed on the sheet is determined entirely by the ViewDefinition that is attached to the sheet - currently, you cannot dynamically override it.

To zoom in on an element in an attached drawing you would need to obtain its Placement in drawing coordinates, transform it into sheet coordinates, and use `Viewport.zoomToPlacements`. To do that you need to know the [Id of the ViewAttachment](https://www.itwinjs.org/reference/core-frontend/locatingelements/viewattachmenthitinfo/) through which the element is being drawn onto the sheet (note, the same drawing element can appear on a single sheet through multiple view attachments), so that you can [obtain the transform](https://www.itwinjs.org/reference/core-frontend/views/viewstate/computedisplaytransform/).

**Reply by fatalstar** [reply]
Thank you for your reply! I will try this solution and provide feedback in here
